### ISSUE TITLE
PANIC: Expected record selection to contain required model ID fields.: ConversionFailure("\"51062868-8142-47da-9ca3-3d90954579e2\"", "Int") in query-engine/core/src/interpreter/interpreter.rs:68:26

### ISSUE THREAD

**Issue by a3diti** [prisma/prisma#12646]
Hi Prisma Team! My Prisma Client just crashed. This is the report:
## Versions

| Name            | Version            |
|-----------------|--------------------|
| Node            | v16.14.2           | 
| OS              | debian-openssl-1.1.x|
| Prisma Client   | 3.11.1             |
| Query Engine    | 0.1.0              |
| Database        | mysql              |

I have crashes when trying to do 4 queries in series,
Most of the errors have this, which says that an **id** format **uuid** type **string** from one table. is being used to convert it for another table that has **id** type **Int** without any sense, as seen on **ConversionFailure**

I checked and rechecked this for some days now, the ID is created by one query, but based on the error, same ID is being used on another query without coding it at all, without specifying it anywhere, the other query is about updating a simple string field on a table without any variable!!

```
PANIC: Expected record selection to contain required model ID fields.: ConversionFailure("\"c756c9c1-6f9b-46fe-83cb-445c530435d5\"", "Int") in query-engine/core/src/interpreter/interpreter.rs:68:26
```

## Logs
<details>
  <summary>
Click to see full log with RUST_BACKTRACE=1
</summary>

```
[1] thread 'tokio-runtime-worker' panicked at 'Expected record selection to contain required model ID fields.: ConversionFailure("\"51062868-8142-47da-9ca3-3d90954579e2\"", "Int")', query-engine/core/src/interpreter/interpreter.rs:68:26
[1] stack backtrace:
[1]    0: rust_begin_unwind
[1]              at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/panicking.rs:498:5
[1]    1: core::panicking::panic_fmt
[1]              at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/panicking.rs:116:14
[1]    2: core::result::unwrap_failed
[1]              at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/result.rs:1690:5
[1]    3: query_core::interpreter::interpreter::ExpressionResult::as_selection_results
[1]    4: core::ops::function::FnOnce::call_once{{vtable.shim}}
[1]    5: query_core::interpreter::interpreter::QueryInterpreter::interpret
[1]    6: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
[1]    7: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
[1]    8: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
[1]    9: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
[1]   10: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
[1]   11: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
[1]   12: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
[1]   13: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
[1]   14: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
[1]   15: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  16: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
[1]   17: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
[1]   18: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
[1]   19: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
[1]   20: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
[1]   21: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
[1]   22: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
[1]   23: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
[1]   24: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
[1]   25: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
[1]   26: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
[1]   27: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
[1]   28: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
[1]   29: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
[1]   30: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
[1]   31: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
[1]   32: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
[1]   33: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
[1]   34: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
[1]   35: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
[1]   36: tokio::runtime::task::harness::poll_future
[1]   37: tokio::runtime::task::harness::Harness<T,S>::poll
[1]   38: std::thread::local::LocalKey<T>::with
[1]   39: tokio::runtime::thread_pool::worker::Context::run_task
[1]   40: tokio::runtime::thread_pool::worker::Context::run
[1]   41: tokio::macros::scoped_tls::ScopedKey<T>::set
  42: tokio::runtime::thread_pool::worker::run
[1]   43: <tokio::runtime::blocking::task::BlockingTask<T> as core::future::future::Future>::poll
[1]   44: tokio::runtime::task::harness::Harness<T,S>::poll
[1]   45: tokio::runtime::blocking::pool::Inner::run
[1] note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
[1] Gabim ne server [register business] PrismaClientUnknownRequestError: 
[1] Invalid `prisma.user.update()` invocation:
[1] 
[1] 
[1]   Expected record selection to contain required model ID fields.: ConversionFailure("\"51062868-8142-47da-9ca3-3d90954579e2\"", "Int")
[1]     at Object.request (/home/dev/projects/se/node_modules/@prisma/client/runtime/index.js:39822:15)
[1]     at PrismaClient._request (/home/dev/projects/se/node_modules/@prisma/client/runtime/index.js:40649:18)
[1]     at Object.callRouteAction (/home/dev/projects/se/node_modules/@remix-run/server-runtime/data.js:40:14)
[1]     at handleDataRequest (/home/dev/projects/se/node_modules/@remix-run/server-runtime/server.js:94:18)
[1]     at requestHandler (/home/dev/projects/se/node_modules/@remix-run/server-runtime/server.js:34:18)
[1]     at /home/dev/projects/se/node_modules/@remix-run/express/server.js:41:22 {
[1]   clientVersion: '3.11.1'
[1] }
[1] POST /regjistro-biznes?_data=routes%2Fuser%2Fregister-business 400 - - 359.268 ms
[1] PrismaClientRustPanicError: 
[1] Invalid `prisma.user.findFirst()` invocation:
[1] 
[1] 
[1]   PANIC: Expected record selection to contain required model ID fields.: ConversionFailure("\"51062868-8142-47da-9ca3-3d90954579e2\"", "Int") in query-engine/core/src/interpreter/interpreter.rs:68:26
[1] 
[1] This is a non-recoverable error which probably happens when the Prisma Query Engine has a panic.
[1] 
[1] https://github.com/prisma/prisma/issues/new?body=Hi+Prisma+Team%21+My+Prisma+Client+just+crashed.+This+is+the+report%3A%0A%23%23+Versions%0A%0A%7C+Name++++++++++++%7C+Version++++++++++++%7C%0A%7C-----------------%7C--------------------%7C%0A%7C+Node++++++++++++%7C+v16.14.2+++++++++++%7C+%0A%7C+OS++++++++++++++%7C+debian-openssl-1.1.x%7C%0A%7C+Prisma+Client+++%7C+3.11.1+++++++++++++%7C%0A%7C+Query+Engine++++%7C+0.1.0++++++++++++++%7C%0A%7C+Database++++++++%7C+mysql++++++++++++++%7C%0A%0A%0A%0A%23%23+Logs%0A%60%60%60%0Aprisma%3AtryLoadEnv+Environment+variables+loaded+from+%2Fhome%2Fdev%2Fprojects%2Fse%2F.env%0Aprisma%3AtryLoadEnv+Environment+variables+loaded+from+%2Fhome%2Fdev%2Fprojects%2Fse%2F.env%0Aprisma%3Aclient+dirname++%2Fhome%2Fdev%2Fprojects%2Fse%2Fnode_modules%2F.prisma%2Fclient%0Aprisma%3Aclient+relativePath++..%2F..%2F..%2Fprisma%0Aprisma%3Aclient+cwd++%2Fhome%2Fdev%2Fprojects%2Fse%2Fprisma%0Aprisma%3Aclient+clientVersion%3A+3.11.1%0Aprisma%3Aclient+clientEngineType%3A+library%0Aprisma%3Aclient%3AlibraryEngine+internalSetup%0Aprisma%3Aclient%3AlibraryEngine+Searching+for+Query+Engine+Library+in+%2Fhome%2Fdev%2Fprojects%2Fse%2Fnode_modules%2F.prisma%2Fclient%0Aprisma%3Aclient%3AlibraryEngine+loadEngine+using+%2Fhome%2Fdev%2Fprojects%2Fse%2Fnode_modules%2F.prisma%2Fclient%2Flibquery_engine-debian-openssl-1.1.x.so.node%0Aprisma%3Aclient%3AlibraryEngine+library+starting%0Aprisma%3Aclient%3AlibraryEngine+library+started%0Aprisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true%0Aprisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true%0Aprisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true%0Aprisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true%0Aprisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true%0Aprisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true%0Aprisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true%0Aprisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true%0Aprisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true%0Aprisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true%0Aprisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true%0Aprisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true%0Aprisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true%0Aprisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true%0Aprisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true%0Aprisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true%0Aprisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true%0Aprisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+true%0A%60%60%60%0A%0A%23%23+Client+Snippet%0A%60%60%60ts%0A%2F%2F+PLEASE+FILL+YOUR+CODE+SNIPPET+HERE%0A%60%60%60%0A%0A%23%23+Schema%0A%60%60%60prisma%0A%2F%2F+PLEASE+ADD+YOUR+SCHEMA+HERE+IF+POSSIBLE%0A%60%60%60%0A%0A%23%23+Prisma+Engine+Query%0A%60%60%60%0A%7B%22X%22%3A%7B%7D%7D%0A%60%60%60%0A&title=PANIC%3A+Expected+record+selection+to+contain+required+model+ID+fields.%3A+ConversionFailure%28%22%5C%2251062868-8142-47da-9ca3-3d90954579e2%5C%22%22%2C+%22Int%22%29+in+query-engine%2Fcore%2Fsrc%2Finterpreter%2Finterpreter.rs%3A68%3A26&template=bug_report.md
[1] 
[1] If you want the Prisma team to look into it, please open the link above 🙏
[1] To increase the chance of success, please post your schema and a snippet of
[1] how you used Prisma Client in the issue. 
[1] 
[1]     at Object.request (/home/dev/projects/se/node_modules/@prisma/client/runtime/index.js:39826:15)
[1]     at PrismaClient._request (/home/dev/projects/se/node_modules/@prisma/client/runtime/index.js:40649:18)
[1]     at isAuthenticated (/home/dev/projects/se/app/utils/session.server.ts:272:19)
[1]     at Object.callRouteLoader (/home/dev/projects/se/node_modules/@remix-run/server-runtime/data.js:77:14)
[1]     at handleDataRequest (/home/dev/projects/se/node_modules/@remix-run/server-runtime/server.js:113:18)
[1]     at requestHandler (/home/dev/projects/se/node_modules/@remix-run/server-runtime/server.js:34:18)
[1]     at /home/dev/projects/se/node_modules/@remix-run/express/server.js:41:22 {
[1]   clientVersion: '3.11.1'
[1] }
```

</details>

## Thoughts
This kind of bug I have seen it on other issues of prisma repo. From today there are **1900 issues** and I believe this repo is not being maintained enough and is being commercialized too much. I am considering switching to another db client after coding for 2 months with Prisma.

**Comment by a3diti** [comment]
Oh yes, just found #11191 similar to this one, with 0 Assignees and 7 participants.

**Comment by a3diti** [comment]
Fixed using **Int** for relation id since using **String** throws this kind of error! Guess this comes from `referentialIntegrity`

**Comment by Weakky** [comment]
Hey folks,

I couldn't reproduce this bug and figured out it was fixed by https://github.com/prisma/prisma-engines/pull/2874 and released in 3.14.

Closing this issue now, please re-open it if you're still experiencing the problem 🙏
